jruby/neo4j/driver/ext/internal_driver.rb

```diff
@@ -11,6 +11,19 @@ module InternalDriver
 
         auto_closable :session
 
+        def execute_query(query, auth_token = nil, config = {}, **parameters)
+          java_method(:executableQuery, [java.lang.String])
+            .call(query)
+            .java_method(:withParameters, [java.util.Map])
+            .call(parameters.transform_keys(&:to_s))
```

Comment:
you need to get hold of the actual java method object only in relatively seldom cases. Mostly the methods can be determined dynamically.
I'm quite sure this will work:
```suggestion
            .with_parameters(to_neo(parameters))
```
please note the conversion `to_neo`. Keys and values have to be converted.